src/business-logic/services/rundown-timeline-service.ts

```diff
@@ -103,15 +111,18 @@ export class RundownTimelineService implements RundownService {
     this.stopAutoNext()
 
     const rundown: Rundown = await this.rundownRepository.getRundown(rundownId)
-
+    const infinitePiecesBeforeStartAutoNext: Map<string,Piece> = rundown.getInfinitePiecesMap()
```

Comment:
`startAutoNext` has nothing to do with updating the infinite Pieces so it shouldn't be in the variable name. It's `rundown.takeNext()` that changes the infinite Pieces of the Rundown.
The `startAutoNext()` just starts a timer, that will execute the `takeNext()` method again which will have its own before and after check for the infinite Pieces.